src/wp-includes/link-template.php

```diff
@@ -1969,9 +1969,15 @@ function get_adjacent_post( $in_same_term = false, $excluded_terms = '', $previo
 	 */
 	$sort = apply_filters( "get_{$adjacent}_post_sort", "ORDER BY p.post_date $order LIMIT 1", $post, $order );
 
-	$query     = "SELECT p.ID FROM $wpdb->posts AS p $join $where $sort";
-	$query_key = 'adjacent_post_' . md5( $query );
-	$result    = wp_cache_get( $query_key, 'counts' );
+	$query        = "SELECT p.ID FROM $wpdb->posts AS p $join $where $sort";
+	$key          = md5( $query );
+	$last_changed = wp_cache_get_last_changed( 'posts' );
+	if ( $in_same_term ) {
```

Comment:
Question: Does this need to be used for excluded terms too? 

I see excluded terms need to be passed as IDs so it's quite possible it's not. Testing for this would be lovely either way though. 

```suggestion
	if ( $in_same_term || ! empty( $excluded_terms ) ) {
```